core/templates/dev/head/editor/RouterServices.js

```diff
@@ -52,8 +52,24 @@ oppia.factory('routerService', [
 
       $rootScope.$broadcast('externalSave');
 
-      if (newPath === '/preview') {
+      if (newPath.indexOf('/preview/') !== -1) {
         _tabs.active = PREVIEW_TAB;
+        var putativeStateName = newPath.substring('/preview/'.length);
+
+        var waitForStatesToLoad = $interval(function() {
```

Comment:
There's duplication here, so I suggest refactoring out the common parts into a separate method.